great_expectations/core/datasource_dict.py

```diff
@@ -77,6 +89,10 @@ def __contains__(self, name: object) -> bool:
     def __setitem__(self, name: str, ds: FluentDatasource | BaseDatasource) -> None:
         config: FluentDatasource | DatasourceConfig
         if isinstance(ds, FluentDatasource):
+            if isinstance(ds, SupportsInMemoryDataAssets):
+                for asset in ds.assets:
```

Comment:
Could we combine the `if` statements?

Reply:
No because it's possible a datasource `SupportsInMemoryDataAssets`, but doesn't actually have any or only some of them are `InMemoryDataAsset`s. We don't want to iterate through all assets unless it makes sense to.